sharding/xorValidatorShuffler_test.go

```diff
@@ -0,0 +1,778 @@
+package sharding
+
+import (
+	"bytes"
+	"crypto/rand"
+	"reflect"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var firstArray = []byte{0xFF, 0xFF, 0xAA, 0xAA, 0x00, 0x00}
+var secondArray = []byte{0xFF, 0x00, 0xAA, 0x55, 0x00, 0xFF}
+var expectedArray = []byte{0x00, 0xFF, 0x00, 0xFF, 0x00, 0xFF}
+
+func generateRandomByteArray(size int) []byte {
+	r := make([]byte, size)
+	_, _ = rand.Read(r)
+
+	return r
+}
+
+func generateValidatorList(number int) []Validator {
+	v := make([]Validator, number)
+
+	for i := 0; i < number; i++ {
+		v[i] = &validator{
+			pubKey: generateRandomByteArray(32),
+		}
+	}
+
+	return v
+}
+
+func generateValidatorMap(
```

Comment:
Suggestion when using random generated data: maybe should add a helper function that display a variadic parameter of  map[uint32][]Validator and use this function as the last parameter in asserts calls. The reason behind this request is: if, by chance, we find (or TC finds) an edge case, we can easily reproduce it because we have the values used. Otherwise, we need to run the same test over and over again hoping that the edge case is reproduced again.